specutils/io/default_loaders/wcs_fits.py

```diff
@@ -171,17 +254,46 @@ def wcs1d_fits_writer(spectrum, file_name, hdu=0, update_header=False, **kwargs)
                        (isinstance(keyword[1], hdr_types) and
                         keyword[0] not in ('NAXIS', 'NAXIS1', 'NAXIS2'))])
 
-    # Cannot include uncertainty in IMAGE_HDU - maybe provide option to
-    # separately write this to BINARY_TBL extension later.
-    if spectrum.uncertainty is not None:
-        warnings.warn("Saving uncertainties in wcs1d format is not yet supported!",
-                      AstropyUserWarning)
-
     # Add flux array and unit
     ftype = kwargs.pop('dtype', spectrum.flux.dtype)
     funit = u.Unit(kwargs.pop('unit', spectrum.flux.unit))
-    flux = spectrum.flux.to(funit, equivalencies=u.spectral_density(wl))
+    flux = spectrum.flux.to(funit, equivalencies=u.spectral_density(disp))
     hdulist[0].data = flux.value.astype(ftype)
+    if flux_name is not None:
+        hdulist[0].name = flux_name
+
+    # Append mask array (duplicate WCS for that extension)?
+    if spectrum.mask is not None and mask_name is not None:
+        hdulist.append(wcs.to_fits()[0])
+        hdulist[-1].data = spectrum.mask
+        if mask_name == '':
+            hdulist[-1].name = 'MASK'
+        else:
+            hdulist[-1].name = mask_name
+        hdu += 1
+    # Warn if extension name for saving was explicitly chosen.
+    elif mask_name is not None and mask_name != '':
+        warnings.warn("No mask found in this Spectrum1D, none saved.", AstropyUserWarning)
+
+    # Append uncertainty array (duplicate WCS for that extension)?
+    if spectrum.uncertainty is not None and uncertainty_name is not None:
```

Comment:
Should we even let user rename the uncertainty HDU? Would be misleading if `spectrum.uncertainty` is STD but then someone names it "IVAR".

Reply:
Good point. There are at least 3 commonly used designations for StdDev, and we probably should also expect things like 'VARIAN' or 'INVVAR' for the others, so if anyone insists they have always used naming convention X in their project and it needs to stay that way, we should not prohibit that. Ultimately should be the responsibility of the user to use sensible values there _if_ they override the defaults, but I should probably add a check that at least it does not conflict with `UNCERT_REF`.

Reply:
Added to tests now, but nonstandard names do complicate everything quite a bit. Wondering if the uncertainty should actually written with the `BUNIT` – examples like MaNGA don't, but it would help with identifying the correct type (and also take care of the case that it might not match the flux unit in the file).

Reply:
Yeah, trying to appease all the random file formats out there is going to be impossible.

Reply:
Agreed, we can only accommodate so much.